File: transport_service_queries/src/main/java/pl/edu/pg/transport/entity/Flight.java
package pl.edu.pg.transport.entity;

import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Document("flights")
public class Flight {

    @Id
    private Long id;

    private String departureAirport;
    private String arrivalAirport;
    private String departureDate;
    private String arrivalDate;
    private int travelTime;
    private int placesCount;
    private int placesOccupied;
    private int price;
}

File: tour_operator/flights/Flight.py
import json


class Flight:
    def __init__(self, flight):
        self.id = None
        if "id" in flight:
            self.id = flight["id"]
        self.departureAirport = flight["departureAirport"]
        self.arrivalAirport = flight["arrivalAirport"]
        self.departureDate = flight["departureDate"]
        self.arrivalDate = flight["arrivalDate"]
        self.travelTime = flight["travelTime"]
        self.placesCount = flight["placesCount"]
        # self.price = flight["price"]
        self.price = 0

    def compare(self, flight_dict):
        flight_dict = json.loads(flight_dict)
        if self.departureAirport != flight_dict["departureAirport"]:
            return False
        if self.arrivalAirport != flight_dict["arrivalAirport"]:
            return False
        if self.departureDate != flight_dict["departureDate"]:
            return False
        if self.arrivalDate != flight_dict["arrivalDate"]:
            return False
        if self.travelTime != flight_dict["travelTime"]:
            return False

        return True
